torch/csrc/jit/export.cpp

```diff
@@ -181,7 +181,28 @@ void encodeGraph(onnx::GraphProto * p_g, const std::shared_ptr<Graph> & g, const
   JIT_ASSERT(p_g != nullptr);
   p_g->set_name("torch-jit-export");
 
+  // The jit tracer may record some unnecessary information.
+  // We should remove the unused input and initializer.
+  // Gather all the names of useful nodes and filter out unused ones.
+  std::unordered_set<std::string> useful_names;
```

Comment:
Why can't you use the `uses()` information here instead of having to do a scan of nodes/outputs first?

Reply:
Because these inputs and initializers are indeed used by a Param node.